R/replay-html.r

```diff
@@ -50,14 +50,32 @@ replay_html.list <- function(x, ...) {
   # keep only high level plots
   parts <- merge_low_plot(parts)
 
-  pieces <- character(length(parts))
+  pieces <- list()
   dependencies <- list()
   for (i in seq_along(parts)) {
     output <- replay_html(parts[[i]], obj_id = i, ...)
     dependencies[[i]] <- attr(output, "knit_meta")
     pieces[[i]] <- output
   }
 
+  # bundle unwrapped pieces into `pre` blocks
+  standalone_piece_indices <- seq_along(pieces) %>%
+    purrr::keep(~ inherits(pieces[[.x]], "html"))
+  split_at_indices <- c(
+    standalone_piece_indices,     # split before html element
+    standalone_piece_indices + 1  # split after html element
+  )
+  bundled_pieces <- pieces %>%
+    split(cumsum(seq_along(pieces) %in% split_at_indices)) %>%
+    purrr::map_if(
+      ~ !inherits(.x[[1]], "html"),
+      ~ paste0("<pre>", paste0(.x, collapse = ""), "</pre>")
+    ) %>%
+    purrr::flatten() %>%
+    purrr::map_chr(as.character) %>%
+    unname()
```

Comment:
I tried the functional approach, which doesn't look great IMO. If I implement it imperatively by looping through elements and merging, the logic is probably even less readable. I can also implement this as `reduce_right` by constantly checking the class of the first element in the accumulator. Let me know if you have any suggestions.